Adapt for usage of the KBUILD_OUTPUT feature.

This feature allows to separate the kernel source from the
kernel build output object files, and the make.conf variable
KBUILD_OUTPUT will be set to the directory containing the kernel
object files (as well as the kernel configuration files) for the
kernel to be used as a target for ebuilds to be emerged.

Note that this can be a different kernel than is currently
running, and it can also be different from the kernel the
/usr/src/linux symlink points to.

Index: etc/make.conf
===================================================================
--- etc.orig/make.conf
+++ etc/make.conf
@@ -465,7 +465,18 @@ DOC_SYMLINKS_DIR="/usr/share/doc/html"
 # kernel build directory (to be used when emerging kernel modules).
 # Note: The /usr/src/linux symlink is *not* used
 # when emerging kernel modules!
-#KBUILD_OUTPUT="/usr/src/linux"
+KBUILD_OUTPUT="/var/tmp/kernel-output/target"
+# Note: The matching initramfs/initrd and built kernel image files can be
+# found at:
+#BOOT_IMAGES="/var/tmp/boot_image/target"
+# (Note that this is not an official Gentoo environment variable, so it is
+# commented out here.)
+# The "target"-path component here is also a symlink pointing to the actual
+# image directory currently used. (There may be multiple versions around.)
+# Hint: *Always* use relative symlinks! In contrary to absolute symlinks,
+# they will continue to work even if the volume containing the partition
+# is mounted into an unfamiliar place. This typically happens during an
+# emergency recovery session launched from some Live CD.
 
 # We consider here those packages using 'GNU Gettext' internally, and for
 # which the installers did not disable translation at _configure_ time. Then,
